File: src/ios/OSInAppBrowserError.swift
enum OSInAppBrowserTarget {
    case openInExternalBrowser
}

enum OSInAppBrowserError: Error {
    case inputArgumentsIssue(target: OSInAppBrowserTarget)
    case openExternalBrowserFailed
    
    private var code: Int {
        var result: Int
        
        switch self {
        case .inputArgumentsIssue: result = 0
        case .openExternalBrowserFailed: result = 0
        }
        
        return result
    }
    
    private var description: String {
        var result: String
        
        switch self {
        case .inputArgumentsIssue: result = ""
        case .openExternalBrowserFailed: result = ""
        }
        
        return result
    }
    
    func toDictionary() -> [String: String] {
        [
            "code": "OS-PLUG-IABP-\(String(format: "%04d", self.code))",
            "message": self.description
        ]
    }   
}

File: src/android/OSInAppBrowserError.kt
package com.outsystems.plugins.inappbrowser.osinappbrowser

enum class OSInAppBrowserError(val code: Int, val message: String) {
    INPUT_ARGUMENTS_ISSUE(100, "The input parameters for 'openInExternalBrowser' are invalid."),
    INPUT_ARGUMENTS_WEB_VIEW_ISSUE(101, "The input parameters for 'openInWebView' are invalid."),
    OPEN_EXTERNAL_BROWSER_FAILED(102, "Couldn't open url using the external browser."),
    OPEN_WEB_VIEW_FAILED(103, "Couldn't open url using the WebView.")
}